plugins/ptp_operator/metrics/metrics.go

```diff
@@ -0,0 +1,381 @@
+	indx := strings.Index(output, "rms")
+	output = output[indx:]
+	fields := strings.Fields(output)
+
+	if len(fields) < 5 {
+		glog.Errorf("%s failed to parse output %s: unexpected number of fields", processName, output)
+		return
+	}
+
+	offsetFromMaster, err := strconv.ParseFloat(fields[1], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse offset from master output %s error %v", processName, fields[1], err)
+	}
+
+	maxOffsetFromMaster, err = strconv.ParseFloat(fields[3], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse max offset from master output %s error %v", processName, fields[3], err)
+	}
+
+	frequencyAdjustment, err = strconv.ParseFloat(fields[5], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse frequency adjustment output %s error %v", processName, fields[5], err)
+	}
+
+	if len(fields) >= 10 {
+		delayFromMaster, err = strconv.ParseFloat(fields[9], 64)
+		if err != nil {
+			glog.Errorf("%s failed to parse delay from master output %s error %v", processName, fields[9], err)
+		}
+	} else {
+		// If there is no delay from master this mean we are out of sync
+		glog.Warningf("no delay from master process %s out of sync", processName)
+	}
+
+	return
+}
+
+func extractRegularMetrics(processName, output string) (offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster float64, clockState ceevent.SyncState) {
+	// remove everything before the rms string
+	// This makes the out to equals
+	output = strings.Replace(output, "path", "", 1)
+	indx := strings.Index(output, "offset")
+	output = output[indx:]
+	fields := strings.Fields(output)
+
+	if len(fields) < 5 {
+		glog.Errorf("%s failed to parse output %s: unexpected number of fields", processName, output)
+		return
+	}
+
+	offsetFromMaster, err := strconv.ParseFloat(fields[1], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse offset from master output %s error %v", processName, fields[1], err)
+	}
+
+	maxOffsetFromMaster, err = strconv.ParseFloat(fields[1], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse max offset from master output %s error %v", processName, fields[1], err)
+	}
+
+	frequencyAdjustment, err = strconv.ParseFloat(fields[4], 64)
+	if err != nil {
+		glog.Errorf("%s failed to parse frequency adjustment output %s error %v", processName, fields[4], err)
+	}
+
+	state := fields[2]
+
+	switch state {
+	case unLocked:
+		clockState = ceevent.FREERUN
+	case clockStep:
+		clockState = ceevent.FREERUN
+	case locked:
+		clockState = ceevent.LOCKED
+	default:
+		glog.Errorf("%s -  failed to parse clock state output `%s` ", processName, fields[2])
+	}
+
+	if len(fields) >= 7 {
+		delayFromMaster, err = strconv.ParseFloat(fields[6], 64)
+		if err != nil {
+			glog.Errorf("%s failed to parse delay from master output %s error %v", processName, fields[6], err)
+		}
+	} else {
+		// If there is no delay from master this mean we are out of sync
+		glog.Warningf("no delay from master process %s out of sync", processName)
```

Comment:
This might spam the logs while you are out of lock.  Could you add a state variable such that you only print on going out of sync?  Also, should this information be incorporated into the state calculation?